src/components/visualizations/chart/highcharts/plugins/group-categories-wrapper.ts

```diff
@@ -16,4 +17,37 @@ export function groupCategoriesWrapper(Highcharts: any) {
 
         hideGridInXAxis.call(this);
     });
+
+    wrap(Highcharts.Tick.prototype, "getPosition", function(
+        proceed: any,
+        horiz: boolean,
+        tickPos: number,
+        tickmarkOffset: number,
+        old: boolean,
+    ) {
+        const {
+            axis: {
+                categories,
+                isXAxis,
+                height,
+                chart: {
+                    options: {
+                        chart: { type },
+                    },
+                },
+            },
+            parent,
+        } = this;
+
+        // for xAxis of bar chart, must update vertical align in the middle
+        // of datalabel for parent category when filtering one value
+        if (isBarChart(type) && isXAxis && parent && this.isNewLabel && categories.length === 1) {
+            parent.labelOffsets.y = height / 2;
+        }
```

Comment:
it could be like below to improve readability :)
```
const hasParent = Boolean(parent);
const hasOneICategory = categories.length === 1;
const shouldAlignMiddle = isBarChart(type) && isXAxis && hasParent && this.isNewLabel && hasOneICategory;

if (shouldAlignMiddle) {...}
```